homeassistant/components/notify/xmpp.py

```diff
@@ -93,29 +118,217 @@ def __init__(self):
             if not verify_certificate:
                 self.add_event_handler('ssl_invalid_cert',
                                        self.discard_ssl_invalid_cert)
+            if data:
+                # Init XEPs for image sending
+                self.register_plugin('xep_0030')  # OOB dep
+                self.register_plugin('xep_0066')  # Out of Band Data
+                self.register_plugin('xep_0071')  # XHTML IM
+                self.register_plugin('xep_0128')  # Service Discovery
+                self.register_plugin('xep_0363')  # HTTP upload
 
             self.connect(force_starttls=self.force_starttls, use_ssl=False)
 
-        def start(self, event):
+        async def start(self, event):
             """Start the communication and sends the message."""
-            self.get_roster()
-            self.send_presence()
+            # Sending image and message independently from each other
+            if data:
+                await self.send_file(timeout=timeout)
+            if message:
+                self.send_text_message()
+
+            self.disconnect(wait=True)
+
+        async def send_file(self, timeout=None):
+            """Send file via XMPP.
+
+            Send XMPP file message using OOB (XEP_0066) and
+            HTTP Upload (XEP_0363)
+            """
             if room:
-                _LOGGER.debug("Joining room %s", room)
                 self.plugin['xep_0045'].join_muc(room, sender, wait=True)
-                self.send_message(mto=room, mbody=message, mtype='groupchat')
+
+            try:
+                # Uploading with XEP_0363
+                _LOGGER.debug("Timeout set to %ss", timeout)
+                url = await self.upload_file(timeout=timeout)
+                if url is None:
+                    raise FileUploadError("Could not upload file")
```

Comment:
Raising that in `upload_file` could simplify the error handling here.

Reply:
yes, that cleans up the code, will do that